rasterio/mask.py

```diff
@@ -3,132 +3,182 @@
 import logging
 import warnings
 
-import rasterio
-from rasterio.features import geometry_mask
+import numpy as np
+
+from rasterio.features import geometry_mask, geometry_window
 
 
 logger = logging.getLogger(__name__)
 
 
-def mask(raster, shapes, nodata=None, crop=False, all_touched=False,
-         invert=False, pad=False):
-    """Mask the area outside of the input shapes with nodata.
+def raster_geom_mask(raster, shapes, all_touched=False, invert=False,
+                    crop=False, pad=False):
+    """Create a mask from shapes, transform, and optional window within original 
+    raster.
 
-    For all regions in the input raster outside of the regions defined by
-    `shapes`, sets any data present to nodata.
+    By default, mask is intended for use as a numpy mask, where pixels that 
+    overlap shapes are False.
+    
+    If shapes do not overlap the raster and crop=True, an exception is 
+    raised.  Otherwise, a warning is raised, and a completely True mask
+    is returned (if invert is False).
 
     Parameters
     ----------
     raster: rasterio RasterReader object
-        Raster to which the mask will be applied.
+        Raster for which the mask will be created.
     shapes: list of polygons
-        Polygons are GeoJSON-like dicts specifying the boundaries of features
-        in the raster to be kept. All data outside of specified polygons
-        will be set to nodata.
-    nodata: int or float (opt)
-        Value representing nodata within each raster band. If not set,
-        defaults to the nodata value for the input raster. If there is no
-        set nodata value for the raster, it defaults to 0.
-    crop: bool (opt)
-        Whether to crop the raster to the extent of the data. Defaults to
-        False.
+        GeoJSON-like dict representation of polygons that will be used to 
+        create the mask.
     all_touched: bool (opt)
         Use all pixels touched by features. If False (default), use only
```

Comment:
I think the verb "use" could be a bit ambiguous here.  How about "Include a pixel in the mask if it touches any of the shapes, If False (default), include a pixel only if its center is within one of the shapes, or if it is selected by Bresenham's line algorithm."